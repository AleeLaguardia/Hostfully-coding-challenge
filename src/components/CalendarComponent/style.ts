import styled from "styled-components";

export const Container = styled.div`
  background-color: ${(props) => props.theme.colors.galery};
  border-radius: 40px;

  .react-calendar {
    display: flex;
    flex-direction: column;
    height: fit-content;
    border-style: none;
    width: 400px;
    border-bottom-left-radius: 40px;
    border-bottom-right-radius: 40px;
    padding: 30px 30px;
    align-items: center;
    border-radius: 40px;

    .react-calendar__navigation__label {
      font-family: ${(props) => props.theme.fonts.poppins};
      color: ${(props) => props.theme.colors.emperor};
      font-weight: 400;
      text-transform: uppercase;
      border: none;
      margin: 0 10px 20px 10px;
      cursor: pointer;

      &:hover {
        background-color: ${(props) => props.theme.colors.galery};
      }
    }

    .react-calendar__navigation__arrow {
      border: none;
      background-color: transparent;
      width: 30px;
      cursor: pointer;
    }

    .react-calendar__month-view__weekdays {
      text-align: center;
    }

    .react-calendar__month-view__weekdays__weekday abbr {
      text-decoration: none;
      font-size: ${(props) => props.theme.sizes.xs};
      color: ${(props) => props.theme.colors.emperor};
      font-family: ${(props) => props.theme.fonts.poppins};
      font-weight: 500;
    }

    .react-calendar__month-view__days abbr {
      text-decoration: none;
      font-size: ${(props) => props.theme.sizes.xs};
      color: ${(props) => props.theme.colors.emperor};
      font-family: ${(props) => props.theme.fonts.poppins};
      font-weight: 500;
    }

    .react-calendar__tile {
      height: 40px;
      width: 15px;
      cursor: pointer;
      border: none;
    }

    .react-calendar__tile--hover {
      background-color: #e6e6e6 !important;
    }

    .react-calendar__tile--hoverEnd {
      border-top-right-radius: 40px;
      border-bottom-right-radius: 40px;
    }

    .react-calendar__tile--hoverStart {
      border-top-left-radius: 40px;
      border-bottom-left-radius: 40px;
    }

    .react-calendar__tile--active,
    .react-calendar__tile--hasActive {
      background-color: ${(props) => props.theme.colors.lightPurple} !important;
    }

    .react-calendar__tile--active abbr,
    .react-calendar__tile--hasActive abbr {
      color: white;
    }

    .react-calendar__tile--hasActive {
      border-radius: 40px;
    }

    .react-calendar__tile--now {
      background-color: purple;

      abbr {
        color: white;
      }
    }

    .react-calendar__tile--now--active {
      background-color: red;
    }

    .react-calendar__tile--rangeBothEnds {
      border-radius: 40px;
    }

    .react-calendar__tile--rangeEnd {
      border-top-right-radius: 40px;
      border-bottom-right-radius: 40px;
      background-color: ${(props) => props.theme.colors.lightPurple};

      abbr {
        color: white;
      }
    }

    .react-calendar__tile--rangeStart {
      border-top-left-radius: 40px;
      border-bottom-left-radius: 40px;
      background-color: ${(props) => props.theme.colors.lightPurple};

      abbr {
        color: white;
      }
    }

    .react-calendar__tile:disabled {
      background-color: transparent;

      abbr {
        color: ${(props) => props.theme.colors.silverChalice};
      }
    }

    .react-calendar__navigation button:disabled {
      background-color: transparent;
      color: ${(props) => props.theme.colors.silverChalice};
    }
  }
`;