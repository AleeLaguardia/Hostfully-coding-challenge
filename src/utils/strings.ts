export const PATH = {
  HOME: '/',
  BOOKING: '/booking',
  RESERVATION: '/reservation',
};

export const HOME = {
  TITLE_FORM: "Book your stay right now!",
  TITLE_PIC: "Book Your Stay",
  ADULTS: "How many adults",
  CHILDREN: "How many children",
}

export const BOOKING = {
  NO_DATA: "No hotels found in that area",
  ALSO_TRY: "But you can also try:"
};