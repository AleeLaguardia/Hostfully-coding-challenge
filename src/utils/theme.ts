import { DefaultTheme } from "styled-components";

import {
  white,
  alabaster,
  emperor,
  galery,
  gray,
  black,
  purple,
  purpleHover,
  darkPurple,
  radicalRed,
  lightPurple,
  silverChalice,
  alto,
  lynch,
  selectiveYellow,
  chelseaGem,
} from './styling/colors';
import {
  poppins,
  playwrite,
} from './styling/fonts';
import {
  xs,
  xxs,
  xxxs,
  s,
  m,
  l,
  xl,
  xxxl,
} from './styling/sizes';

export const theme: DefaultTheme = {
  colors: {
    white,
    alabaster,
    emperor,
    galery,
    gray,
    black,
    purple,
    purpleHover,
    darkPurple,
    radicalRed,
    lightPurple,
    silverChalice,
    alto,
    lynch,
    selectiveYellow,
    chelseaGem,
  },
  fonts: {
    poppins,
    playwrite,
  },
  sizes: {
    xs,
    xxs,
    xxxs,
    s,
    m,
    l,
    xl,
    xxxl,
  },
}