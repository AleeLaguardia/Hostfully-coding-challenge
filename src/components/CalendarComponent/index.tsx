import React from "react";
import { useDispatch, useSelector } from "react-redux";
import Calendar from "react-calendar";
import { Container } from "./style";
import { setUser } from "../../store/slice/userInfoSlice";
import { RootState } from "../../store";
import { Value } from "react-calendar/dist/cjs/shared/types";

interface Props {
  isOpen: boolean;
  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
}

const CalendarComponent: React.FC<Props> = ({ isOpen, setIsOpen }) => {
  const dispatch = useDispatch();
  const user = useSelector((state: RootState) => state.user);

  const handleDateChange = (date: Value) => {
    if (isOpen) {
      console.log("Date selected:", date); // Log the selected date
      setIsOpen(false);
      dispatch(setUser({ ...user, date }));
    }
  };

  const tileDisabled = ({ date }: { date: Date }) => {
    const today = new Date();
    today.setHours(0, 0, 0, 0);
    return date < today;
  };

  return (
    <Container data-testid="calendar-component">
      <Calendar
        tileClassName="tile-container"
        selectRange
        locale="en-GB"
        className="calendar-container"
        returnValue="range"
        showNeighboringMonth={false}
        onChange={(date: Value) => handleDateChange(date)}
        value={user.date[0] || user.date[1] ? user.date : [new Date(), new Date()]}
        tileDisabled={tileDisabled}
        prev2Label={false}
        next2Label={false}
        view="month"
      />
    </Container>
  );
};

export default CalendarComponent;
