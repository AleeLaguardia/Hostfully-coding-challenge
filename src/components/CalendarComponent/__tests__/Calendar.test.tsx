import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import CalendarComponent from '..';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../../utils/theme';

const mockStore = configureStore([]);

describe('CalendarComponent', () => {
  let store: any;
  let setIsOpen: jest.Mock;

  beforeEach(() => {
    store = mockStore({
      user: {
        destination: 'Paris',
        adults: '2 adults',
        children: '1 child',
        date: [new Date('2024-07-14'), new Date('2024-07-17')],
      },
    });
    setIsOpen = jest.fn();
    store.dispatch = jest.fn();
  });

  const renderWithProviders = (ui: React.ReactElement) => {
    return render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          {ui}
        </ThemeProvider>
      </Provider>
    )
  }

  it('renders CalendarComponent with default styling', () => {
    const { asFragment } = renderWithProviders(<CalendarComponent isOpen={true} setIsOpen={setIsOpen} />);

    expect(asFragment).toMatchSnapshot();
  });

  it('renders CalendarComponent correctly', () => {
    const { getByText } = renderWithProviders(<CalendarComponent isOpen={true} setIsOpen={setIsOpen} />);

    expect(getByText('July 2024')).toBeInTheDocument();
  });

  it('handles date change correctly', () => {
    const { getByText } = renderWithProviders(<CalendarComponent isOpen={true} setIsOpen={setIsOpen} />);

    fireEvent.click(getByText('15'));
    fireEvent.click(getByText('17'));
  });
});
