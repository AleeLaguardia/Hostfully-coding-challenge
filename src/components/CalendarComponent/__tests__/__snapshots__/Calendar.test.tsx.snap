// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarComponent renders CalendarComponent with default styling 1`] = `[Function]`;
